### PR TITLE
Cmr 3932 collection citation mapping iso smap

### PR DESCRIPTION
**Pull request by sxu123**

### CODE REVIEW

File: umm-spec-lib/src/cmr/umm_spec/test/iso_shared.clj [hunk 1]
```diff
@@ -1,17 +1,60 @@
 (ns cmr.umm-spec.test.iso-shared
   "ISO 19115 specific expected conversion functionality"
   (:require
-    [clojure.string :as string]
+    [clojure.string :as str]
```
Comment (laurenfrederick) [inline umm-spec-lib/src/cmr/umm_spec/test/iso_shared.clj]: Why did you change this? we've been moving towards string.

File: umm-spec-lib/src/cmr/umm_spec/test/iso_smap_expected_conversion.clj [hunk 2]
```diff
@@ -277,6 +278,20 @@
        iso-shared/sort-by-date-type-iso
        (#(or (seq %) (not-provided-temporal-extents)))))
 
+(defn- expected-collection-citations
+  "Returns collection-citations with only the first collection-citation in it, trimmed,
+   and if the Title is nil, replace it with Not provided.
+   When collection-citations is nil, return [{:Title \"Not provided\"}]."
```
Comment (laurenfrederick) [inline umm-spec-lib/src/cmr/umm_spec/test/iso_smap_expected_conversion.clj]: This is a helpful comment.

File: umm-spec-lib/src/cmr/umm_spec/test/iso_shared.clj (hunk 1 above)
Comment (mschmele) [inline umm-spec-lib/src/cmr/umm_spec/test/iso_shared.clj]: Nitpick: prefer `string` over `str`

Comment (sxu123) [inline umm-spec-lib/src/cmr/umm_spec/test/iso_shared.clj]: Done.

Comment (sxu123) [inline umm-spec-lib/src/cmr/umm_spec/test/iso_shared.clj]: Done.

File: umm-spec-lib/src/cmr/umm_spec/test/iso_shared.clj [hunk 3]
```diff
@@ -1,17 +1,60 @@
...
+         :FirstName (:FirstName person-names)
+         :MiddleName (:MiddleName person-names)
+         :LastName (:LastName person-names)}))))
+
+(defn update-contact-persons-from-collection-citation
+  "CollectionCitation introduces additional contact persons from creator, editor and publisher.
+   They need to be added to the ContactPersons in the expected umm after converting the umm
+   to the xml and back to umm.  Returns the updated ContactPersons."
+  [contact-persons collection-citation]
+  (let [creator (:Creator collection-citation)
```
Comment (sxu123) [inline umm-spec-lib/src/cmr/umm_spec/test/iso_shared.clj]: Done.

File: umm-spec-lib/src/cmr/umm_spec/test/iso_shared.clj [hunk 4]
```diff
@@ -1,17 +1,60 @@
...
+        publisher-contact-person (create-contact-person publisher)]
+    (remove nil?  (conj contact-persons
+                        (util/remove-nil-keys creator-contact-person)
+                        (util/remove-nil-keys editor-contact-person)
+                        (util/remove-nil-keys publisher-contact-person)))))
+
+(defn trim-collection-citation
+  "Returns CollectionCitation with Creator, Editor, Publisher and ReleasePlace fields trimmed."
+  [collection-citation]
+  (let [creator (:Creator collection-citation)
```
Comment (sxu123) [inline umm-spec-lib/src/cmr/umm_spec/test/iso_shared.clj]: Done.
